### DISCUSSION TITLE
Beanie is returning No ID, just like new document even I searched by object ID

### DISCUSSION THREAD

**Discussion by ccppoo** [BeanieODM/beanie#1013] [Question]
sorry for no real code, I cant' reproduce it

I inherit class, `DocumentBase` like this:

```python
from datetime import datetime
from typing import List, Optional, Union
from beanie import Document, Link
from pydantic import Field, BaseModel, ConfigDict, AliasGenerator
from pydantic.alias_generators import to_camel, to_snake
from app.utils.time import datetime_utc
from pprint import pprint

from beanie.odm.actions import after_event
from beanie.odm.actions import Insert, Save, Replace, Update, SaveChanges


class DocumentBase(Document):
    """DocumentBase"""

    uploaded_at: datetime = Field(default_factory=datetime_utc)
    last_edited: Optional[datetime] = Field(default=None)

    model_config = ConfigDict(
        alias_generator=AliasGenerator(
            validation_alias=to_snake, serialization_alias=to_camel
        )
    )

    @property
    def id_str(self) -> str:
        return str(self.id)

    @after_event(Replace, Update, SaveChanges)
    def update_last_edit(self):
        self.last_edited = datetime_utc()
```

after adding new model, it's returning `id='None'`

even I search document like

```python
manu = await Manufacturer.get("66d001e9f2d9ae267ee19172")
```

**Comment by ccppoo** [comment]
I just thought `model_config` works because beanie is dependent to pydantic, my mistake

don't define `model_config` in `Document`
```python

model_config = ConfigDict(
        alias_generator=AliasGenerator(
            validation_alias=to_snake, serialization_alias=to_camel
        )
    )
```

**Reply by stopffen** [reply]
so how did you manage to get your alias serialization to work using beanie? 
Im looking for a solution like this, since i use snake_case in python but camel in my db.

**Reply by ccppoo** [reply]
> so how did you manage to get your alias serialization to work using beanie? Im looking for a solution like this, since i use snake_case in python but camel in my db.


I uploaded this when I tried to work with single beanie ORM class with interacting HTTP frontend and also with backend, DB.

So there are 2 case when validating comes in

1. reading from frontend
2. reading from mongoDB

serializing and validation problem happens when

1. making new instance from backend side,
2. new instance from frontend request

if frontend use camel-case  but snake-case at backend, the problem happens

I could have make classmethod for making instance using snake-case, but that made to much redundancy

---

I just made another dataclass that corresponds with frontend model (`JSON`, `Form`)

and made `classmethod` in Beanie ORM class that returns raw `dict` and revalidate with `dataclass`

beanie ORM -> between fastapi and mongo DB
dataclass -> between fastapi and frontend

in this way, it was much clear how I could handle frontend circumstances seperated from schema design

like returning as sorted array instead of key-value object format when frontend asked for.

--- 

I started making backend and frontend at the same time, so when I needed new JSON schema for some data like I mentioned above

It was much better way to handle with frontend requirements without manipulating MongoDB schema

**Reply by stopffen** [reply]
hi thanks for your answer!

Yes I also do already use two model between Frontend/Backend and Backend/Mongo. 
Are you now using camelCase in your beanie classes, cause I see that this would solve my issue but i wanna stick to snake_case in my backend for consistency. 

Maybe we have a slightly different setup. I want to use camelCase in Mongo and the Fronted, while snake_case in the backend

**Reply by ccppoo** [reply]
> hi thanks for your answer!
> 
> Yes I also do already use two model between Frontend/Backend and Backend/Mongo. Are you now using camelCase in your beanie classes, cause I see that this would solve my issue but i wanna stick to snake_case in my backend for consistency.
> 
> Maybe we have a slightly different setup. I want to use camelCase in Mongo and the Fronted, while snake_case in the backend

in that case, `validation_alias=to_snake, serialization_alias=to_camel` would be the case 

MongoDB (camel) ---(validation)---> Beanie(snake)

Beanie(snake) ---(serialize)---> Mongo(camel)

I follow same naming convention for DB and backend, because sometimes I need to write raw queries.

So I'm not really sure about your case.

good luck
